packages/pnpm/src/main.ts

```diff
@@ -109,6 +108,10 @@ export default async function run (inputArgv: string[]) {
     process.exitCode = 1
     return
   }
+  if (!config.useBetaCli) {
+    process.env['npm_config_argv'] = JSON.stringify(argv)
+    config.rawConfig.argv = process.env['npm_config_argv']
```

Comment:
ok I figured out that the lifecycle scripts rely on `config.rawConfig.argv` for their `process.env['npm_config_argv']`, and this is previously set at `@pnpm/config`